### ISSUE TITLE
c++ load python pytorch model, forward result is different?

### ISSUE THREAD

**Issue by nickhuangxinyu** [pytorch/pytorch#57819]
I have a trained model in pytorch, i saved it using:

    tsm = torch.jit.trace(mlp, torch.rand(1, num_features).double())
    tsm.save('model.pt')
    print(mlp(np.array([[0.2] * num_features])))

and i load it in c++ and forward using the same data -> [0.2] * num_features, code is following:

    #include <iostream>
    #include <memory>
    #include <torch/script.h> // One-stop header.
    #include <torch/custom_class.h>
    
    int main(int argc, const char* argv[]) {
      if (argc != 2) {
        std::cerr << "usage: example-app <path-to-exported-script-module>\n";
        return -1; 
      }
    
      torch::jit::script::Module module = torch::jit::load(argv[1]);
    
      std::vector<torch::jit::IValue> inputs;
      // std::vector<double> inputs;                                                                                                                                                                                                            
      // inputs.push_back(torch::ones({1, 51}));
      std::vector<double> v(51, 0.2);
      torch::Tensor data = torch::from_blob(v.data(), {1, 51});
      inputs.push_back(data);
    
      // at::Tensor output = module.forward(inputs).toTensor();
      // std::cout << output.slice(/*dim=*/1, /*start=*/0, /*end=*/5) << '\n';
      auto t = module.forward(inputs).toTensor();
      std::cout << t.sizes() << std::endl;
      std::cout << module.forward(inputs).toTensor()[0][0].item<double>();//.toDouble() << std::endl;
    }

The output is:

`0.112908%` and `tensor([0.0010], dtype=torch.float64, grad_fn=<AddBackward0>)`

can anyone help on this? why the result is different?

cc @gmagogsfm

**Comment by nickhuangxinyu** [comment]
the reason is double and float in c++
